ledger/eval/txntracer.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package eval
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+)
+
+var txnGroupDeltaTracer *TxnGroupDeltaTracer
+
+// TODO do we want something like this?
+type txnGroupDeltas struct {
+	// Ids contains all the associated IDs for these changes (group and txn IDs)
+	Ids []crypto.Digest
+	ledgercore.StateDelta
+}
+
+// TxnGroupDeltaTracer collects groups of StateDelta objects covering groups of txns
+type TxnGroupDeltaTracer struct {
+	// lookback is the number of rounds stored at any given time
+	Lookback uint64
```

Comment:
why is this the only exported field?

Reply:
Good point, I don't think it needs to be.